include/extractor/guidance/node_based_graph_walker.hpp

```diff
@@ -105,10 +105,35 @@ struct SelectRoadByNameOnlyChoiceAndStraightness
      */
     boost::optional<EdgeID> operator()(const NodeID nid,
                                        const EdgeID via_edge_id,
-                                       const Intersection &intersection,
+                                       const IntersectionView &intersection,
+                                       const util::NodeBasedDynamicGraph &node_based_graph) const;
+
+    const NameID desired_name_id;
+    const bool requires_entry;
```

Comment:
Hrm here and below: the selectors / accumulators are structs with all member attributes public. But from what I can tell here, these are completely useless as the callsite sets them in the ctor.

What about at least making internals private and public what the user should be able to access?